grails-app/controllers/au/org/ala/biocache/hubs/OccurrenceController.groovy

```diff
@@ -57,38 +62,25 @@ class OccurrenceController {
      */
     def list(SpatialSearchRequestParams requestParams) {
         def start = System.currentTimeMillis()
-        requestParams.fq = params.list("fq") as String[] // override Grails binding which splits on internal commas in value
-
-        log.debug "requestParams = ${requestParams}"
-
-        List taxaQueries = (ArrayList<String>) params.list("taxa") // will be list for even one instance
-        log.debug "skin.useAlaBie = ${grailsApplication.config.skin.useAlaBie}"
-        log.debug "taxaQueries = ${taxaQueries} || q = ${requestParams.q}"
 
-        if (grailsApplication.config.skin.useAlaBie?.toString()?.toBoolean() &&
-                grailsApplication.config.bie.baseUrl && taxaQueries && taxaQueries[0]) {
-            // check for list with empty string
-            // taxa query - attempt GUID lookup
-            List guidsForTaxa = webServicesService.getGuidsForTaxa(taxaQueries)
-            def additionalQ = (params.q) ? " AND " + params.q : "" // advanced search form can provide both taxa and q params
-            requestParams.q = postProcessingService.createQueryWithTaxaParam(taxaQueries, guidsForTaxa) + additionalQ
-        } else if (!params.q && taxaQueries && taxaQueries[0]) {
-            // Bypass BIE lookup and pass taxa query in as text
-            List emptyGuidList = taxaQueries.clone().collect { it = ""} // list of empty strings, of equal size to taxaQueries
-            requestParams.q = postProcessingService.createQueryWithTaxaParam(taxaQueries, emptyGuidList)
-        }
+        def activeProfile = time("active profile") { qualityService.activeProfile(requestParams.qualityProfile) }
 
-        if (!requestParams.q) {
-            requestParams.q = "*:*"
-        }
+        normaliseRequestParams(requestParams)
 
         try {
             //the configured grouping
-            Map configuredGroupedFacets = webServicesService.getGroupedFacets()
-            List listOfGroupedFacets = postProcessingService.getListFromGroupedFacets(configuredGroupedFacets)
-            Map defaultFacets = postProcessingService.getAllFacets(listOfGroupedFacets)
-            String[] userFacets = postProcessingService.getFacetsFromCookie(request)
-            String[] filteredFacets = postProcessingService.getFilteredFacets(defaultFacets)
+            Map configuredGroupedFacets
+            List listOfGroupedFacets
+            Map defaultFacets
+            String[] userFacets
+            String[] filteredFacets
+            time("all facets") {
```

Comment:
It will be triggered in production, it doesn't need to be wrapped, the overhead is minimal.

IMHO `time` is pretty obvious, it times the following code block.

Reply:
Yeah, once I read further, its clear enough - although it comes across as a fair bit of "code pollution" as the wrapper code makes readability a bit worse, TBH. I suppose we can live with it - code is near EOL anyway.